src/modules/solution-explorer/solution-explorer-solution/delete-diagram-modal/delete-diagram-modal.ts

```diff
@@ -60,22 +77,52 @@ export class DeleteDiagramModal {
 
   private _closeModal(): void {
     this.diagram = undefined;
-    this._solutionService = undefined;
+    this._solutionExplorerService = undefined;
 
     this.showModal = false;
   }
 
   private async _deleteDiagram(): Promise<void> {
     try {
-      await this._solutionService.deleteDiagram(this.diagram);
+      await this._solutionExplorerService.deleteDiagram(this.diagram);
     } catch (error) {
       const message: string = `Unable to delete the diagram: ${error.message}`;
 
       this._notificationService.showNotification(NotificationType.ERROR, message);
     }
 
+    const diagramIndex: number = this._openDiagramService
+      .getOpenedDiagrams()
+      .findIndex((diagram: IDiagram) => diagram.uri === this.diagram.uri);
+
+    const searchIndex: number = diagramIndex === 0 ? diagramIndex + 1 : diagramIndex - 1;
```

Comment:
```suggestion
    const previousOrNextDiagramIndex: number = diagramIndex === 0 ? diagramIndex + 1 : diagramIndex - 1;
```
würde direkt erklären wofür der index steht.